crates/youki/src/logger.rs

```diff
@@ -36,12 +46,12 @@ pub fn init(log_file: Option<PathBuf>, log_format: Option<String>) -> Result<()>
     } else {
         env_logger::Target::Stderr
     };
-    env_logger::Builder::from_env(
-        env_logger::Env::default().filter_or("YOUKI_LOG_LEVEL", DEFAULT_LOG_LEVEL),
-    )
-    .format(formatter)
-    .target(target)
-    .init();
+    env_logger::Builder::new()
+        .parse_env(env_logger::Env::new().filter("YOUKI_LOG_LEVEL"))
+        .filter_level(FromStr::from_str(log_level).unwrap())
+        .format(formatter)
+        .target(target)
+        .init();
```

Comment:
This doesn't give a clear idea of which level has the highest priority.
So it is better to decide which log level you want in lines 33-37 and just set the value in the builder and do not overwrite it.

Reply:
This thing is starting to get tricky for me

Reply:
What areas became tricky?

Reply:
Get the value from the environment variable and convert it to the appropriate type

Reply:
> Get the value from the environment variable and convert it to the appropriate type

It didn't seem like it would be a very complex code, but what are your concerns?

Reply:
Doesn't this code overwrite the default level even if `YOUKI_LOG_LEVEL` is set?

Reply:
> > Get the value from the environment variable and convert it to the appropriate type
> 
> It didn't seem like it would be a very complex code, but what are your concerns?

Get the value from the environment variable and convert it to &str, output a friendly error if it fails

Reply:
```rust
let log_level = if log_debug_flag {
        "debug"
    } else {
        match env::var("YOUKI_LOG_LEVEL") {
            Ok(val) => 
            Err(_e) => DEFAULT_LOG_LEVEL
        }
    };
```